### DISCUSSION TITLE
Qusetions about Tone Label and the CLI API

### DISCUSSION THREAD

**Discussion by rael346** [ChenglongMa/SkinToneClassifier#12] [Q&A] [answered]
Hi,

Currently, I'm planning to use this for a school project but couldn't understand the `tone label` column in the result csv. Here are some questions I have:
- Based on the release log, it was previously PERLA, so is the underlying classification still using the PERLA scale?
- In the command line, there is a `-l` flag for having custom label. The first letter in the default is the image type, but what is the second letter (Since PERLA only has 11 categories whereas the last letter goes from A - Z)?
```
  -l LABEL [LABEL ...], --labels LABEL [LABEL ...]
                        Skin tone labels; default values are the uppercase alphabet list leading by the image type ('C' for 'color'; 'B' for 'Black&White'), e.g., ['CA', 'CB', ..., 'CZ'] or ['BA', 'BB', ..., 'BZ'].
```
- I'm planning to use a different color palette like the **The Fitzpatrick Skin Scale**, but I'm not entirely sure how the `--palette` and `--label` works and what I'm supposed to give it.

**Comment by ChenglongMa** [comment]
Hi @rael346,

Thanks for your feedback and sorry for the confusion.

For your questions,
1. Yes, the default value of palette is still PERLA scale. I just change its name.
2. The `label` is just a "name" for a "color" in the `palette`. So the algorithm requires a palette containing color values, like RGB or HEX values.
3. Since The Fitzpatrick Skin Scale does not have direct RGB or HEX values associated with it. I'm afraid it can't be used directly for this project.

Please let me know if I can help you for your project. 

Thanks.

**Reply by rael346** [reply]
so how do I interpret the tone label of an image? Say the label is CI, I know that the first letter is C, then what is `I` in terms of the PERLA scale?

**Reply by ChenglongMa** [reply]
Hi @rael346,

Yes, correct.

If you are using the default palette, then "C" for "Color", "I" is the category in PERLA scale.

Thanks!

**Reply by rael346** [reply]
So based on [this link](https://rpubs.com/Haibiostat/skintonepalette) I found, the PERLA color palette is numerical. I couldn't find other source that mentions the alphabetical value. Can you link me to the source of the palette that you are basing on?

**Reply by ChenglongMa** [reply, marked as answer]
Thanks for the link!

In [our paper](https://doi.org/10.1111/ssqu.13242) mentioned in the README, we just map the numerical values to alphabetical letters. For example, "1" -> "A", "2" -> "B",..., "7" -> "G".

If you want to use the numerical labels, you can specify it by:

```bash
stone -i /path/to/images -d -l 1 2 3 4 5 6 7 8 9 10 11
```

Thanks.
